crates/matrix-sdk-crypto/src/session_manager/sessions.rs

```diff
@@ -414,18 +414,23 @@ impl SessionManager {
                 .collect();
 
             if !missing_devices.is_empty() {
+                let mut missing_devices_by_user: BTreeMap<_, BTreeSet<_>> = BTreeMap::new();
+                for (user_id, device_id) in missing_devices.into_iter() {
+                    missing_devices_by_user
+                        .entry(user_id)
+                        .or_default()
+                        .insert((*device_id).clone());
+                }
+
                 warn!(
-                    ?missing_devices,
-                    "Tried to create a new sessions, but the signed one-time key was missing for some devices",
+                    ?missing_devices_by_user,
+                    "Tried to create new Olm sessions, but the signed one-time key was missing for some devices",
                 );
```

Comment:
Logging the missing devices in this way means that we don't have to repeat the user id for each device, and generally makes it (a bit) easier to read.